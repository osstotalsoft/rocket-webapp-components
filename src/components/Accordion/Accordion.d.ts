export interface AccordionProps {
    /**
    * @default -1
    * The index of the active child.
    */
    active?: number;
    /**
     * The content of accordion.
     */
    content: {
        title: string,
        content: React.ReactNode
    }[];
}
/**
 *
 * Demos:
 *
 * - https://bit.dev/totalsoft_oss/react-mui/accordion
 *
 * - Take a look over the code for further information: https://github.com/osstotalsoft/rocket-webapp-components/blob/master/components/Accordion/Accordion.js
 * 
 * Material-UI components used:
 * 
 * - Accordion: https://material-ui.com/api/accordion/
 * - AccordionSummary: https://material-ui.com/api/accordion-summary/
 * - AccordionDetails: https://material-ui.com/api/accordion-details/
 * 
 */
export default function Accordion(props: AccordionProps): JSX.Element;