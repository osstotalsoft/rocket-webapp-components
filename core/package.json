{
  "engines": {
    "node": ">=10"
  },
  "scripts": {
    "prebuild": "rimraf build",
    "build": "npm run build:cjs &&  npm run build:copy-files",
    "build:cjs": "babel src -d build --copy-files",
    "build:copy-files": "node ./scripts/copy-files.js",
    "bit-build": "bit -v && bit import && bit build"
  },
  "devDependencies": {
    "@babel/cli": "^7.8.0",
    "@babel/core": "^7.7.7",
    "@babel/node": "^7.7.7",
    "@babel/plugin-proposal-decorators": "^7.8.3",
    "@babel/preset-env": "^7.7.7",
    "@babel/preset-react": "^7.8.0",
    "babel-eslint": "^10.0.3",
    "cross-env": "^7.0.2",
    "eslint": "^6.8.0",
    "eslint-config-prettier": "^6.7.0",
    "eslint-config-standard": "^14.1.0",
    "eslint-config-standard-react": "^9.2.0",
    "eslint-plugin-import": "^2.18.2",
    "eslint-plugin-node": "^11.0.0",
    "eslint-plugin-prettier": "^3.1.1",
    "eslint-plugin-promise": "^4.2.1",
    "eslint-plugin-react": "^7.17.0",
    "eslint-plugin-standard": "^4.0.1",
    "gh-pages": "^2.2.0",
    "npm-run-all": "^4.1.5",
    "prettier": "^1.19.1",
    "react": "^16.13.1",
    "react-dom": "^16.13.1",
    "react-scripts": "^3.4.1",
    "rimraf": "^3.0.0",
    "url-search-params-polyfill": "^7.0.1"
  },
  "peerDependencies": {
    "react": "^16.13.1",
    "react-dom": "^16.13.1",
    "react-flags": "^0.1.18",
    "react-router-dom": "^5.1.2"
  },
  "sideEffects": false,
  "dependencies": {},
  "bit": {
    "env": {
      "compiler": "totalsoft_oss.react-mui/compilers/react@1.0.32"
    },
    "componentsDefaultDirectory": "components/{name}",
    "packageManager": "npm"
  }
}
