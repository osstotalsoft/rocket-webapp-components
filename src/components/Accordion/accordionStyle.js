import styles from '../../assets/jss/styles';

const accordionStyle = theme => {

  const { themeColor } = styles(theme);

  return {
    root: {
      flexGrow: 1,
      marginBottom: "20px",
      '@media (max-width: 480px)': {
        width: "100%"
      },

    },
    expansionPanel: {
      boxShadow: "none",
      "&:before": {
        display: "none !important"
      }
    },
    expansionPanelExpanded: {
      margin: "0"
    },
    expansionPanelSummary: {
      minHeight: "auto !important",
      backgroundColor: "transparent",
      // borderBottom: "1px solid #ddd",
      padding: "2px 2px 2px 0px",
      borderTopLeftRadius: "3px",
      borderTopRightRadius: "3px",
      color: "#3C4858",
      "&:hover": {
        color: themeColor
      }
    },
    expansionPanelSummaryExpaned: {
      color: themeColor
    },
    expansionPanelSummaryContent: {
      margin: "0"
    },
    expansionPanelSummaryExpandIcon: {
      [theme.breakpoints.up("md")]: {
        top: "auto !important"
      },
      transform: "rotate(0deg)",
      color: "inherit",
      [theme.breakpoints.down("sm")]: {
        top: "10px !important"
      }
    },
    expansionPanelSummaryExpandIconExpanded: {
      [theme.breakpoints.up("md")]: {
        top: "auto !important"
      },
      transform: "rotate(180deg)",
      [theme.breakpoints.down("sm")]: {
        top: "10px !important"
      }
    },
    title: {
      fontSize: "15px",
      fontWeight: "bolder",
      marginTop: "0",
      marginBottom: "0",
      color: "inherit"
    },
    expansionPanelDetails: {
      padding: "15px 0px 5px"
    }
  }
};

export default accordionStyle;