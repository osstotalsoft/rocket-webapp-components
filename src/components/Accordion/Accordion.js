import React, { useState } from "react";
import PropTypes from "prop-types";
import { Accordion as AccordionBase, AccordionSummary, AccordionDetails, makeStyles } from "@material-ui/core";
import ExpandMore from "@material-ui/icons/ExpandMore";
import accordionStyle from "./accordionStyle";
const useStyles = makeStyles(accordionStyle);

const Accordion = ({ content, active }) => {

    const [localActive, setLocalActive] = useState(active)
    const handleChange = panel => (_, expanded) => setLocalActive(expanded ? panel : -1)
    const classes = useStyles();

    return (
        
        <div className={classes.root}>
            {content.map((prop, key) => {
                return (
                    <AccordionBase
                        expanded={localActive === key}
                        onChange={handleChange(key)}
                        key={key}
                        classes={{
                            root: classes.expansionPanel,
                            expanded: classes.expansionPanelExpanded
                        }}
                    >
                        <AccordionSummary
                            expandIcon={<ExpandMore />}
                            classes={{
                                root: classes.expansionPanelSummary,
                                expanded: classes.expansionPanelSummaryExpaned,
                                content: classes.expansionPanelSummaryContent,
                                expandIcon: classes.expansionPanelSummaryExpandIcon,
                                //expandIconExpanded: classes.expansionPanelSummaryExpandIconExpanded
                            }}
                        >
                            <h4 className={classes.title}>{prop.title}</h4>
                        </AccordionSummary>
                        <AccordionDetails className={classes.expansionPanelDetails}>
                            {prop.content}
                        </AccordionDetails>
                    </AccordionBase>
                );
            })}
        </div>
    );
}


Accordion.defaultProps = {
    active: -1
};

Accordion.propTypes = {
    /**
    * The index of the active child.
    */
    active: PropTypes.number,
    /**
    * The content of accordion.
    */
    content: PropTypes.arrayOf(
        PropTypes.shape({
            title: PropTypes.string,
            content: PropTypes.node
        })
    ).isRequired
};

export default Accordion;