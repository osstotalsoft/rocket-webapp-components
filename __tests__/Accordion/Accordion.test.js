import React from 'react';
import Accordion from '../../components/Accordion/Accordion';
import { shallow, mount } from "enzyme";
import { Accordion as AccordionBase } from "@material-ui/core";

const someArray = [{
    title: 'Accordion 1',
    value: 'Client 1'
}, {
    title: 'Accordion 2',
    value: 'Client 2'
},
{
    title: 'Accordion 3',
    value: 'Client 3'
}
]

describe('Accordion', () => {
    it('Three children are expected to render', () => {
        const wrapper = shallow(<Accordion
            active={0}
            content={someArray.map(item => {
                return {
                    title: item.title,
                    content: (
                        <div>
                            {item.value}
                        </div>
                    )
                };
            })}
        />);

        expect(wrapper.find(AccordionBase)).toHaveLength(3);
    });

    it('Second child is expanded', () => {
        const wrapper = mount(<Accordion
            active={1}
            content={someArray.map(item => {
                return {
                    title: item.title,
                    content: (
                        <div>
                            {item.value}
                        </div>
                    )
                };
            })}
        />);

        expect(wrapper.prop('active')).toBe(1)
        expect(wrapper.find(AccordionBase).at(1).props().expanded).toBe(true)
    });

    it('Third child is expanded', () => {
        const wrapper = mount(<Accordion
            active={2}
            content={someArray.map(item => {
                return {
                    title: item.title,
                    content: (
                        <div>
                            {item.value}
                        </div>
                    )
                };
            })}
        />);

        expect(wrapper.find(AccordionBase).at(2).props().expanded).toBe(true)
    });
});