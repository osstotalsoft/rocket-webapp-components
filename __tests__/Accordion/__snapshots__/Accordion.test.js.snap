// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion First child is expanded 1`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="MuiPaper-root MuiAccordion-root makeStyles-expansionPanel-2 Mui-expanded makeStyles-expansionPanelExpanded-3 MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      aria-disabled={false}
      aria-expanded={true}
      className="MuiButtonBase-root MuiAccordionSummary-root makeStyles-expansionPanelSummary-4 Mui-expanded makeStyles-expansionPanelSummaryExpaned-5"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiAccordionSummary-content makeStyles-expansionPanelSummaryContent-6 Mui-expanded makeStyles-expansionPanelSummaryExpaned-5"
      >
        <h4
          className="makeStyles-title-9"
        >
          Accordion 1
        </h4>
      </div>
      <div
        aria-disabled={false}
        aria-hidden={true}
        className="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon makeStyles-expansionPanelSummaryExpandIcon-7 Mui-expanded makeStyles-expansionPanelSummaryExpaned-5 MuiIconButton-edgeEnd"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role={null}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </div>
    </div>
    <div
      className="MuiCollapse-root MuiCollapse-entered"
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper"
      >
        <div
          className="MuiCollapse-wrapperInner"
        >
          <div
            role="region"
          >
            <div
              className="MuiAccordionDetails-root makeStyles-expansionPanelDetails-10"
            >
              <div>
                Client 1
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiPaper-root MuiAccordion-root makeStyles-expansionPanel-2 MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      aria-disabled={false}
      aria-expanded={false}
      className="MuiButtonBase-root MuiAccordionSummary-root makeStyles-expansionPanelSummary-4"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiAccordionSummary-content makeStyles-expansionPanelSummaryContent-6"
      >
        <h4
          className="makeStyles-title-9"
        >
          Accordion 2
        </h4>
      </div>
      <div
        aria-disabled={false}
        aria-hidden={true}
        className="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon makeStyles-expansionPanelSummaryExpandIcon-7 MuiIconButton-edgeEnd"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role={null}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </div>
    </div>
    <div
      className="MuiCollapse-root MuiCollapse-hidden"
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper"
      >
        <div
          className="MuiCollapse-wrapperInner"
        >
          <div
            role="region"
          >
            <div
              className="MuiAccordionDetails-root makeStyles-expansionPanelDetails-10"
            >
              <div>
                Client 2
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiPaper-root MuiAccordion-root makeStyles-expansionPanel-2 MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      aria-disabled={false}
      aria-expanded={false}
      className="MuiButtonBase-root MuiAccordionSummary-root makeStyles-expansionPanelSummary-4"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiAccordionSummary-content makeStyles-expansionPanelSummaryContent-6"
      >
        <h4
          className="makeStyles-title-9"
        >
          Accordion 3
        </h4>
      </div>
      <div
        aria-disabled={false}
        aria-hidden={true}
        className="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon makeStyles-expansionPanelSummaryExpandIcon-7 MuiIconButton-edgeEnd"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role={null}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </div>
    </div>
    <div
      className="MuiCollapse-root MuiCollapse-hidden"
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper"
      >
        <div
          className="MuiCollapse-wrapperInner"
        >
          <div
            role="region"
          >
            <div
              className="MuiAccordionDetails-root makeStyles-expansionPanelDetails-10"
            >
              <div>
                Client 3
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Accordion Third child is expanded 1`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="MuiPaper-root MuiAccordion-root makeStyles-expansionPanel-2 MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      aria-disabled={false}
      aria-expanded={false}
      className="MuiButtonBase-root MuiAccordionSummary-root makeStyles-expansionPanelSummary-4"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiAccordionSummary-content makeStyles-expansionPanelSummaryContent-6"
      >
        <h4
          className="makeStyles-title-9"
        >
          Accordion 1
        </h4>
      </div>
      <div
        aria-disabled={false}
        aria-hidden={true}
        className="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon makeStyles-expansionPanelSummaryExpandIcon-7 MuiIconButton-edgeEnd"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role={null}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      className="MuiCollapse-root MuiCollapse-hidden"
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper"
      >
        <div
          className="MuiCollapse-wrapperInner"
        >
          <div
            role="region"
          >
            <div
              className="MuiAccordionDetails-root makeStyles-expansionPanelDetails-10"
            >
              <div>
                Client 1
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiPaper-root MuiAccordion-root makeStyles-expansionPanel-2 MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      aria-disabled={false}
      aria-expanded={false}
      className="MuiButtonBase-root MuiAccordionSummary-root makeStyles-expansionPanelSummary-4"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiAccordionSummary-content makeStyles-expansionPanelSummaryContent-6"
      >
        <h4
          className="makeStyles-title-9"
        >
          Accordion 2
        </h4>
      </div>
      <div
        aria-disabled={false}
        aria-hidden={true}
        className="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon makeStyles-expansionPanelSummaryExpandIcon-7 MuiIconButton-edgeEnd"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role={null}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      className="MuiCollapse-root MuiCollapse-hidden"
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper"
      >
        <div
          className="MuiCollapse-wrapperInner"
        >
          <div
            role="region"
          >
            <div
              className="MuiAccordionDetails-root makeStyles-expansionPanelDetails-10"
            >
              <div>
                Client 2
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiPaper-root MuiAccordion-root makeStyles-expansionPanel-2 Mui-expanded makeStyles-expansionPanelExpanded-3 MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      aria-disabled={false}
      aria-expanded={true}
      className="MuiButtonBase-root MuiAccordionSummary-root makeStyles-expansionPanelSummary-4 Mui-expanded makeStyles-expansionPanelSummaryExpaned-5"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiAccordionSummary-content makeStyles-expansionPanelSummaryContent-6 Mui-expanded makeStyles-expansionPanelSummaryExpaned-5"
      >
        <h4
          className="makeStyles-title-9"
        >
          Accordion 3
        </h4>
      </div>
      <div
        aria-disabled={false}
        aria-hidden={true}
        className="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon makeStyles-expansionPanelSummaryExpandIcon-7 Mui-expanded makeStyles-expansionPanelSummaryExpaned-5 MuiIconButton-edgeEnd"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role={null}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      className="MuiCollapse-root MuiCollapse-entered"
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper"
      >
        <div
          className="MuiCollapse-wrapperInner"
        >
          <div
            role="region"
          >
            <div
              className="MuiAccordionDetails-root makeStyles-expansionPanelDetails-10"
            >
              <div>
                Client 3
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Accordion Three children are expected to render 1`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="MuiPaper-root MuiAccordion-root makeStyles-expansionPanel-2 Mui-expanded makeStyles-expansionPanelExpanded-3 MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      aria-disabled={false}
      aria-expanded={true}
      className="MuiButtonBase-root MuiAccordionSummary-root makeStyles-expansionPanelSummary-4 Mui-expanded makeStyles-expansionPanelSummaryExpaned-5"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiAccordionSummary-content makeStyles-expansionPanelSummaryContent-6 Mui-expanded makeStyles-expansionPanelSummaryExpaned-5"
      >
        <h4
          className="makeStyles-title-9"
        >
          Accordion 1
        </h4>
      </div>
      <div
        aria-disabled={false}
        aria-hidden={true}
        className="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon makeStyles-expansionPanelSummaryExpandIcon-7 Mui-expanded makeStyles-expansionPanelSummaryExpaned-5 MuiIconButton-edgeEnd"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role={null}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      className="MuiCollapse-root MuiCollapse-entered"
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper"
      >
        <div
          className="MuiCollapse-wrapperInner"
        >
          <div
            role="region"
          >
            <div
              className="MuiAccordionDetails-root makeStyles-expansionPanelDetails-10"
            >
              <div>
                Client 1
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiPaper-root MuiAccordion-root makeStyles-expansionPanel-2 MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      aria-disabled={false}
      aria-expanded={false}
      className="MuiButtonBase-root MuiAccordionSummary-root makeStyles-expansionPanelSummary-4"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiAccordionSummary-content makeStyles-expansionPanelSummaryContent-6"
      >
        <h4
          className="makeStyles-title-9"
        >
          Accordion 2
        </h4>
      </div>
      <div
        aria-disabled={false}
        aria-hidden={true}
        className="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon makeStyles-expansionPanelSummaryExpandIcon-7 MuiIconButton-edgeEnd"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role={null}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      className="MuiCollapse-root MuiCollapse-hidden"
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper"
      >
        <div
          className="MuiCollapse-wrapperInner"
        >
          <div
            role="region"
          >
            <div
              className="MuiAccordionDetails-root makeStyles-expansionPanelDetails-10"
            >
              <div>
                Client 2
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiPaper-root MuiAccordion-root makeStyles-expansionPanel-2 MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      aria-disabled={false}
      aria-expanded={false}
      className="MuiButtonBase-root MuiAccordionSummary-root makeStyles-expansionPanelSummary-4"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiAccordionSummary-content makeStyles-expansionPanelSummaryContent-6"
      >
        <h4
          className="makeStyles-title-9"
        >
          Accordion 3
        </h4>
      </div>
      <div
        aria-disabled={false}
        aria-hidden={true}
        className="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon makeStyles-expansionPanelSummaryExpandIcon-7 MuiIconButton-edgeEnd"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role={null}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      className="MuiCollapse-root MuiCollapse-hidden"
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper"
      >
        <div
          className="MuiCollapse-wrapperInner"
        >
          <div
            role="region"
          >
            <div
              className="MuiAccordionDetails-root makeStyles-expansionPanelDetails-10"
            >
              <div>
                Client 3
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
